docs/contributing/provider-design.md

```diff
@@ -0,0 +1,133 @@
+
+Please note: the overall design and implementation across all AWS functionality is federated. This means that individual services may implement concepts and use terminology differently. As such, this guide may not be fully exhaustive in all situations. The goal is to provide enough hints about these concepts and towards specific terminology to point contributors in the correct direction, especially when researching prior implementations.
+
+### Authorization and Acceptance Resources
+
+Certain AWS services use an authorization-acceptance model for cross-account associations or access. Some examples of these can be found with:
+
+* Direct Connect Association Proposals
+* GuardDuty Member Invitations
+* RAM Resource Share Associations
+* Route 53 VPC Associations
+* Security Hub Member Invitations
+
+This type of API model either implements a form of an invitation/proposal identifier that can be used to accept the authorization, otherwise it may just be a matter of providing the desired AWS Account identifier as the target of an authorization. In the latter case, the acceptance is implicit when creating the other half of the association.
+
+To model the separate invitation/proposal in Terraform resources:
+
+* Typically the authorization side has separate and sufficient creation and read API functionality to create an "invite" or "proposal" resource.
+* If the acceptance side has separate and sufficient accept, read, and reject API functionality, then an "accepter" resource may be created, where the operations are mapped as:
+    * Create: Accepts the invitation/proposal.
+    * Read: Reads the invitation/proposal to determine status. If not found, then it should fallback to reading the API resource associated with the invitation/proposal/association. As evidenced in some previous API implementations, the invitation/proposal may be temporary and removed from the API after an indeterminate amount of time which may not be documented or easily discoverable from testing.
+    * Delete: Rejects or otherwise deletes the invitation/proposal.
+
+Otherwise, to model an implicit acceptance in Terraform resources:
+
+* If the authorization side has separate and sufficient creation and read API functionality, an "authorization" resource is created.
+* An "association" resource is created, where the operations are mapped as:
```

Comment:
Are these 2 sentences trying to say the same thing?

Reply:
I'll combine these into the above "Otherwise" sentence -- they are different, but my writing was unclear.

Reply:
IMHO it's more understandable now that the relationship between association and authorization is spelled out.